bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java

```diff
@@ -289,12 +298,13 @@ public void addInboxListener(InboxListener listener) throws IllegalStateExceptio
     }
 
     @Override
+    @NonNullByDefault({})
     public Stream<DiscoveryResult> stream() {
-        return this.discoveryResultStorage.getValues().stream();
+        return this.discoveryResultStorage.getValues().stream().filter(discoveryResult -> discoveryResult == null);
```

Comment:
if you filter that way you will end up in streaming nulls, won't you? Wasn't your idea to filter `discoveryResult != null` ?

Reply:
`getValues()` is declarated that it returns a collection that contains potential null entries. So without the filter this method needs to be changed to a stream that contains nullable entries.

Reply:
@maggu2810 Yes, but still the filter should be the other way around: `discoveryResult != null`.

Reply:
You are surely right :rofl: